framework/markdown.go

```diff
@@ -0,0 +1,124 @@
+/*
+ Copyright 2025 The OSCAL Compass Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package framework
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"strings"
+	"text/template"
+
+	oscalTypes "github.com/defenseunicorns/go-oscal/src/types/oscal-1-1-2"
+)
+
+//go:embed template/*.md
+var embeddedResources embed.FS
+
+type TemplateValues struct {
+	Catalog           string
+	Component         string
+	AssessmentResults oscalTypes.AssessmentResults
+}
+
+// Get the catalog title as the template.md catalog info
+func getCatalogTitle(catalog oscalTypes.Catalog) (string, error) {
+	if catalog.Metadata.Title != "" {
+		return catalog.Metadata.Title, nil
+	} else {
+		return "", fmt.Errorf("Error getting catalog title")
+	}
+}
+
+// Get the component title as the template.md component info
+// At that stage, it only supports the Components length is 1. When the
+// observation links to the component in assessment plan, it will be improved.
+func getComponentTitle(assessmentPlan oscalTypes.AssessmentPlan) (string, error) {
+	if len(*assessmentPlan.LocalDefinitions.Components) == 1 {
```

Comment:
A nil check would be good here as well since `LocalDefinitions` could be nil.

Reply:
@jpower432 thanks, updated.